### PR TITLE
Kube - support List documents

### PR DESCRIPTION
**Pull request by ygalblum**
Flatten List into documents
Add List type to meta/v1
Add unittest

#### Does this PR introduce a user-facing change?
Yes

```release-note
Kube - add support for List kind
```

Resolves: #19052

### CODE REVIEW

**Review by giuseppe — APPROVED** [review]
LGTM

File: pkg/domain/infra/abi/play.go [hunk 1]
```diff
@@ -1261,13 +1262,35 @@ func splitMultiDocYAML(yamlContent []byte) ([][]byte, error) {
...
+		if err != nil {
+			return nil, fmt.Errorf("individual doc yaml could not be marshalled: %w", err)
+		}
+
+		kind, err := getKubeKind(document)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't get object kind: %w", err)
+		}
+
+		if kind == "List" {
```
Comment (vrothberg) [inline pkg/domain/infra/abi/play.go:1280]: Can you add a comment why lists need special casing?

**Review by vrothberg — COMMENTED** [review]
I lack understanding on how we want this to behave on the command level.  Running the example YAML for the unit test gives me `Error: pod does not have a name`.  Can we add an e2e test?

**Comment by ygalblum** [comment]
> Will do

Done

**Review by vrothberg — APPROVED** [review]
LGTM, thank you!

File: pkg/domain/infra/abi/play.go (hunk 1 above)
Comment (ygalblum) [reply pkg/domain/infra/abi/play.go:1282]: Done

**Pull request merged**